src/syft/core/nodes/abstract/worker.py

```diff
@@ -86,3 +93,60 @@ def _register_services(self) -> None:
                     self.msg_router[handler_type_subclass] = service_instance
 
         self.services_registered = True
+
+    @type_hints
+    def _fetch_network(self) -> None:
```

Comment:
This is the wrong location for this I imagine. Do we also want to allow VMs, Devices, and Domains all to do this? I think you probably want this functionality on the Domain object which is the only thing which is "network aware" quite intentionally. The reason is because the Domain is responsible for making sure that no private information is leaked to the network.

Reply:
Putting this on the "Worker" class means that all nodes will have it... meaning every VM, and every Device, and every Domain.

Reply:
Putting this on the "Worker" class means that all nodes will have it... meaning every VM, and every Device, and every Domain.

Reply:
Even Network nodes themselves would also have it.

Reply:
A network for a VM could be as simple as "who is my device?"
A network for a network could be "what's my public route and any routes that I'm allowed to be aware of"

Reply:
Ah - ok that's cool. Let's not overload the word "network" though. Let's find another word.